### ISSUE TITLE
New gatsby version 2.23.3 every page = Loading (StaticQuery)

### ISSUE THREAD

**Issue by apmsooner** [gatsbyjs/gatsby#24890]
Just updated gatsby to latest version and this is all i see on every page now. I assume related to this: https://github.com/gatsbyjs/gatsby/pull/24805

**Comment by nonAlgebraic** [comment]
Same here, but I experience this on Gatsby v.2.22.15, too.

I've narrowed it down to version 2.9.5 of `babel-plugin-remove-graphql-queries` - at version 2.9.4 it's all good, one patch version up and all pages with a `StaticQuery` fail.

**Comment by LekoArts** [comment]
Hi!

Sorry to hear you're running into an issue. To help us best begin debugging the underlying cause, it is incredibly helpful if you're able to create a [minimal reproduction][reproduction]. This is a simplified example of the issue that makes it clear and obvious what the issue is and how we can begin to debug it.

If you're up for it, we'd very much appreciate if you could [provide a minimal reproduction][reproduction] and we'll be able to take another look.

Thanks for using Gatsby! 💜

[reproduction]: https://gatsby.dev/reproduction

**Comment by elrumordelaluz** [comment]
I found the same issue but it was enough removing `node_modules && yarn.lock` and regenerating _packages_ doing `yarn`.

**Comment by sidharthachatterjee** [comment]
Hey folks! Thanks for opening this. I just tried this out and couldn't reproduce this with latest gatsby and 2.9.5 of `babel-plugin-remove-graphql-queries`. 

Does anyone have a project that reproduces this consistently? 

I suspect this might have to do with babel loader cache (which is in `node_modules/.cache`) so please try to clear that folder along with `gatsby clean` and try again! 

Thanks!

**Comment by elrumordelaluz** [comment]
> I suspect this might have to do with babel loader cache (which is in node_modules/.cache) so please try to clear that folder along with gatsby clean and try again!

same here, since it worked everything after clean `node_modules` and `yarn.lock`

**Comment by bmatzner** [comment]
Can confirm this in a site that uses StaticQuery in an SEO component  (very much along the lines of https://www.gatsbyjs.org/docs/use-static-query/) – forcing babel-plugin-remove-graphql-queries to v2.9.4 does indeed resolve the issue

**Comment by atomicpages** [comment]
OMG sorry I keep closing the issue -- I've legit been working a lot on RTL UIs recently... I keep switching buttons in my head. Anyway, can someone post a diff of their lockfile? That might help narrow down possible culprits. I still haven't been able to reproduce the issue

**Comment by briangonzalezmia** [comment]
Thought I was going crazy! Deleting node_modules and lock file did it for me too. Thank you!!

**Comment by LekoArts** [comment]
Since this can't be reproduced on a fresh install and all comments suggest that cleaning ` node_modules/.cache` solves the issue I don't think there is much we can do here.

So please clean that folder (and maybe a `gatsby clean` for good measure). If you're on e.g. Netlify you'll also need to re-run with `Clean cache & build` step. Sorry for the troubles!

**Comment by mminute** [comment]
Seeing this issue on a project I started using `gatsby-starter-hello-world`.  I've finally finished (everything looks great with `gatsby develop`!), but when I run `gatsby build` and `gatsby serve` I get this output `Loading (StaticQuery)`.  I'm using a `<StaticQuery/>` to load images.  I've tried `gatsby clean` and `rm -rf node_modules` then `npm install` many times but I still see the issue.

The only thing that has worked for me is @nonAlgebraic 's solution of updating `package-lock.json` for `babel-plugin-remove-graphql-queries` to version `2.9.4` instead of `2.9.5`.  The only issue that I'm running into with this solution is that using `BackgroundImage` from 'gatsby-background-image' is broken.

Gatsby version 2.23.3
repo: https://github.com/mminute/masonjenningsIOv2
